addons/JobChange/jobchange.lua

```diff
@@ -27,119 +27,181 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+    elseif command ~= nil and command ~= '' then
+        main = command:lower()
+        if main and main:upper() == p.main_job then main = nil end
+    else
+        log('Syntax: //jc main|sub JOB  -- Chnages main or sub to target JOB')
+        log('Syntax: //jc main/sub  -- Changes main and sub')
+        log('Syntax: //jc reset -- Resets Current Job')
+        return
+    end
+
+    local changes = T{}
+
+    local main_id = find_job(main, p)
+    if main ~= nil and main_id == nil then 
+        log('Could not change main job to to '..main:upper()..' ---Mistype|NotUnlocked')
+        return
+    end
+    local sub_id = find_job(sub, p)
+    if sub ~= nil and sub_id == nil then 
+        log('Could not change sub job to to '..sub:upper()..' ---Mistype|NotUnlocked')
+        return
+    end
+
+    if main_id == nil and sub_id == nil then
+        log('No change required.')
+        return
+    end
+
+    if main_id ~= nil and main:upper() == p.sub_job then
+        if sub_id ~= nil and sub:upper() == p.main_job then
+            changes:append({job_id=find_temp_job(p), is_conflict=true, is_main=false})
+            changes:append({job_id=main_id, is_main=true})
+            changes:append({job_id=sub_id, is_main=false})
+        else
+            if sub_id ~= nil then
+                changes:append({job_id=sub_id, is_main=false})
+            else
+                changes:append({job_id=find_temp_job(p), is_conflict=true, is_main=false})
+            end
+            changes:append({job_id=main_id, is_main=true})
+        end
+    elseif sub_id ~= nil and sub:upper() == p.main_job then
+        if main_id ~= nil then
+            changes:append({job_id=main_id, is_main=true})
+        else
+            changes:append({job_id=find_temp_job(p), is_conflict=true, is_main=true})
+        end
+        changes:append({job_id=sub_id, is_main=false})
+    else
+        if main_id ~= nil then
+            if main:upper() == p.main_job then
+                changes:append({job_id=find_temp_job(p), is_conflict=true, is_main=true})
+            end
+            changes:append({job_id=main_id, is_main=true})
+        end
+        if sub_id ~= nil then
+            if sub:upper() == p.sub_job then
+                changes:append({job_id=find_temp_job(p), is_conflict=true, is_main=false})
+            end
+            changes:append({job_id=sub_id, is_main=false})
+        end
+    end
+
+    local npc = find_job_change_npc()
+    if npc then
+        for _, change in ipairs(changes) do
+            if change.is_conflict then
+                log('Conflict with '..(change.is_main and 'main' or 'sub')..' job. Changing to: '..res.jobs[change.job_id].ens)
+            else
+                log('Changing '..(change.is_main and 'main' or 'sub')..' job to: '..res.jobs[change.job_id].ens)
+            end
+            jobchange(change.job_id, change.is_main)
+
+            coroutine.sleep(0.5)
+        end
+    else
+        log('Not close enough to a Moogle!')
```

Comment:
This case could be checked for at the very start to save some otherwise unnecessary calculations.